osquery/tables/system/darwin/user_groups.mm

```diff
@@ -151,24 +160,28 @@ QueryData genUsers(QueryContext& context) {
         genODEntries(kODRecordTypeUsers, username, usernames);
 
         Row r;
-        r["username"] = pwd->pw_name;
-        r["is_hidden"] = INTEGER(usernames[r["username"]]);
         genUserRow(r, pwd);
+        r["is_hidden"] = INTEGER(usernames[r["username"]]);
         results.push_back(r);
       }
     } else {
       std::map<std::string, bool> usernames;
       genODEntries(kODRecordTypeUsers, nil, usernames);
       for (const auto& username : usernames) {
+        // opendirectory and getpwnam are documented as having
+        // different code paths. Thus we may see cases where
+        // genODEntries produces responses that are not in
+        // getpwnam. So with a surfeit of caution we populate some of
+        // the row here
+        Row r;
+        r["username"] = TEXT(username.first.c_str());
```

Comment:
You moved setting “username” into the genUserRow method but not “is_hidden”.

Reply:
It was intentional.

the genUserRow (and genGroupRow) are meant to act on the return from `getpwuid` In contrast, `is_hidden` comes from the opendirectory lookup and is less normalized.

Reply:
Ok, let's get this merged, one more follow up and then I'm GTG!

In this case (with users) and in the case above (with groups) a successful POSIX API call will overwrite a value, either `username` or `groupname`. Can we be super specific and instead of overwriting the value, handle the caution with an `} else { ... }`?